### DISCUSSION TITLE
A Simple Method to Automatically Back Up Snapshots (Tabs, Windows) to the Cloud (e.g., Dropbox)

### DISCUSSION THREAD

**Discussion by davidleejy** [mbnuqw/sidebery#1497] [General]
Hi fellow Sidebery users, 

Through this write-up, I'd like to share a simple method I discovered for backing up Firefox tabs and windows as Sidebery snapshots to the cloud. (A snapshot is a record of opened Firefox tabs and windows)

![asdfasdf](https://github.com/mbnuqw/sidebery/assets/1690072/d6d0c3e7-cbf7-4717-b52b-32e946621520)

### Motivation

As at writing this in early 2024, backing up snapshots isn't trivial because: 

- Sidebery can only automatically create snapshots in the Downloads directory which is separate from directories that cloud services are synchronizing. As such, snapshots being created in this manner aren't synchronized to any cloud.

- Sidebery's Sync isn't able to sync tabs & windows. It only syncs settings, context menu settings, styles, and keybindings (to the user's Firefox profile).

_Aside_ from scheduling a chron job, a simple and elegant method to synchronize tabs and windows I discovered is as follows.

### Method

The main idea behind my method is to have Sidebery automatically create snapshots at regular intervals in a location that's really a softlink to a directory residing in your cloud directory.

Here are the steps:

**Step 1**. Create a directory in your cloud directory that is intended to house snapshots. I named this directory `sidebery-snapshots`. You may name this however you please.

**Step 2**. Create a softlink in the Downloads directory that's linked to `sidebery-snapshots` by executing this terminal command:
```bash
$ ln -s /Users/d/Dropbox/sidebery-snapshots /Users/d/Downloads/sidebery-snapshots-do-not-delete
```
Executing this terminal command creates a softlink named `sidebery-snapshots-do-not-delete` in the `Downloads` directory that's linked to `sidebery-snapshots` directory.

**Step 3**. Access Sidebery's settings to configure Sidebery to (automatically) create snapshots in `~/Downloads/sidebery-snapshots-do-not-delete` at regular intervals. This step is illustrated in this screenshot:

![Screenshot 2024-03-02 at 12 43 45 AM](https://github.com/mbnuqw/sidebery/assets/1690072/56d2f598-64e9-4eb4-a34b-ae8f0913dd5d)

And that's it, we're done. 😃 

Now, whenever Sidebery writes files to directory `sidebery-snapshots-do-not-delete` (located in `Downloads`), these files are in fact written to directory `Dropbox/sidebery-snapshots` which is constantly being synchronized to my Dropbox cloud.

This method should work for other cloud service providers (e.g., Windows OneDrive, Google Cloud).

**Comment by nuf0n** [comment]
Legendary solution, thank you!!!
